hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/webapp/FederationInterceptorREST.java

```diff
@@ -1172,7 +1172,28 @@ public ApplicationStatisticsInfo getAppStatistics(HttpServletRequest hsr,
   @Override
   public NodeToLabelsInfo getNodeToLabels(HttpServletRequest hsr)
       throws IOException {
-    throw new NotImplementedException("Code is not implemented");
+    NodeToLabelsInfo nodeToLabelsInfo = new NodeToLabelsInfo();
+    Map<SubClusterId, SubClusterInfo> subClustersActive;
+    try {
+      subClustersActive = getActiveSubclusters();
```

Comment:
You could do this in a single try, and catch the exception there.
There is no best way to return the error than an empty NodeToLabelsInfo?